### PR TITLE
e2e tests for products page

### LINKED ISSUE #181: End to end (e2e) test for products page
Implement e2e tests for products page, test sorting and pagination

### PR DESCRIPTION
**Pull request by heshvab**
- Added products page feature file
- Added cypress tests for it

Since we use our mock backend, I doubt there is sense in checking right sorting order of products by comparing first and last items as I previously told. Instead, I added checking for request search parameters that come from client to the server when client fetches products.

### CODE REVIEW

File: e2e/cypress/test_module/cypress/support/step_definitions/products-page.steps.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
+/// <reference types="Cypress" />
+
+import { Given, Then, When } from "@badeball/cypress-cucumber-preprocessor";
+import { sortingNameParamMap } from "@cypress-e2e/fixtures/constants";
+
+Given("The user is on the products page", () => {
+  cy.visit("http://localhost:3000/products");
```
Comment (OliaHrytsak) [inline e2e/cypress/test_module/cypress/support/step_definitions/products-page.steps.ts]: You can change path on "/products"

File: e2e/cypress/test_module/cypress/support/step_definitions/products-page.steps.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+});
+
+Given("The user is on the first page of products", () => {
+  cy.visit("http://localhost:3000/products?page=1");
+  cy.intercept("GET", /\/api\/v1\/products/).as("productsPageRequest");
+});
+
+Given("The user is on the second page of products", () => {
+  cy.visit("http://localhost:3000/products?page=2");
+  cy.intercept("GET", /\/api\/v1\/products/).as("productsPageRequest");
```
Comment (OliaHrytsak) [inline e2e/cypress/test_module/cypress/support/step_definitions/products-page.steps.ts]: We have global-data.ts with constants httpMethod. You should use them

File: e2e/cypress/test_module/cypress/support/step_definitions/products-page.steps.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+When("The user looks at the sorting dropdown", () => {
+  cy.getById("products-dropdown").should("be.visible");
+});
+
+When("The user opens sorting dropdown", () => {
+  cy.getById("products-dropdown").click();
+});
+
+When("The user clicks {int} page", (page: number) => {
+  cy.intercept("GET", new RegExp("page=" + page)).as("productsPageRequest");
```
Comment (OliaHrytsak) [inline e2e/cypress/test_module/cypress/support/step_definitions/products-page.steps.ts]: use constant also there

**Review by mhevyk — APPROVED** [review]

**Review by D1nviz — APPROVED** [review]

**Review by dmtrth25 — APPROVED** [review]

File: e2e/cypress/test_module/cypress/support/step_definitions/products-page.steps.ts (hunk 3 above)
Comment (heorhiiss) [reply e2e/cypress/test_module/cypress/support/step_definitions/products-page.steps.ts]: Thank you, I used constants here and in other places

File: e2e/cypress/test_module/cypress/support/step_definitions/products-page.steps.ts (hunk 2 above)
Comment (heorhiiss) [reply e2e/cypress/test_module/cypress/support/step_definitions/products-page.steps.ts]: Thank you, fixed

File: e2e/cypress/test_module/cypress/support/step_definitions/products-page.steps.ts (hunk 1 above)
Comment (heorhiiss) [reply e2e/cypress/test_module/cypress/support/step_definitions/products-page.steps.ts]: Thanks for suggestion, fixed

**Review by OliaHrytsak — APPROVED** [review]

**Pull request merged**